jupyterhub/app.py

```diff
@@ -2048,6 +2048,8 @@ def cleanup_oauth_clients(self):
     def init_proxy(self):
         """Load the Proxy config"""
         # FIXME: handle deprecated config here
+        if self.ssl_cert and self.ssl_key:
+            self.bind_url = self.bind_url.replace('http', 'https')
```

Comment:
to avoid replacing 'http' elsewhere in the url:

```suggestion
            self.bind_url = self.bind_url.replace('http://', 'https://', 1)
```

Reply:
Ooohh, hadn't thought about it! Good spot!